core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -1791,4 +1793,33 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     assertTrue(records2.count() == 1 && records2.records(tp).asScala.head.offset == 1,
       "Expected consumer2 to consume one message from offset 1, which is the committed offset of consumer1")
   }
+  @Test
+  def testStaticConsumerDetectsNewPartitionCreatedAfterRestart(): Unit = {
```

Comment:
I didn't see any consumer restart in the test. Are you trying to say `afterNewConsumerCreated`?

Reply:
Well.. We can't really restart a consumer, right? The only way is to recreate the consumer. What I meant here is that the static member is restarted.

Reply:
Make sense. Thanks.